src/QuizList.js

```diff
@@ -13,11 +15,47 @@ const customHash = input => {
   return hash;
 };
 
-const QuizList = ({ history, setQuiz }) => {
-  const [categories, setCategories] = useState([]);
-  const [background] = useState(getColor);
+const PublicQuizList = () => {
+  const history = useHistory();
+  const [quizzes, setQuizzes] = useState(null);
+  const [error, setError] = useState(false);
+
+  if (!quizzes) {
+    const db = firebase.firestore();
+    db.collection("publicQuizzes")
+      .doc("list")
+      .get()
+      .then(doc =>
+        doc.exists ? setQuizzes(Object.entries(doc.data())) : setError(true)
+      )
+      .catch(() => setError(true));
+  }
 
-  if (categories.length === 0) {
+  return (
+    <>
+      <h2>Public Quizzes (you can request for your quiz to become public)</h2>
+      {error && <span>ERROR Loading quiz list.</span>}
+      {!error && !quizzes && <span>Loading quizzes . . .</span>}
+      {!error &&
+        quizzes &&
+        quizzes.map(([id, name]) => (
+          <button
+            className="fullwidth-button"
+            key={name}
+            onClick={() => history.push(`/q/${id}`)}
+          >
+            {name}
+          </button>
+        ))}
+    </>
+  );
+};
+
+const OpenTDbList = ({ setQuiz }) => {
```

Comment:
Maybe in the future there should be also a URN for OpenTDb quizzes, and we could avoid the {setQuiz} variable.